resilience4j-micronaut/src/main/java/io/github/resilience4j/mirconaut/annotation/Bulkhead.java

```diff
@@ -0,0 +1,49 @@
+package io.github.resilience4j.mirconaut.annotation;
+
+import io.github.resilience4j.mirconaut.bulkhead.BulkheadInterceptor;
+import io.github.resilience4j.mirconaut.bulkhead.ThreadPoolBulkheadInterceptor;
+import io.micronaut.aop.Around;
+import io.micronaut.context.annotation.Type;
+
+import java.lang.annotation.*;
+
+/**
+ * This annotation can be applied to a class or a specific method. Applying it on a class is
+ * equivalent to applying it on all its public methods.
+ */
+@Retention(value = RetentionPolicy.RUNTIME)
+@Target(value = {ElementType.METHOD, ElementType.TYPE})
+@Around
+@Type({BulkheadInterceptor.class, ThreadPoolBulkheadInterceptor.class})
+@Documented
+public @interface Bulkhead {
```

Comment:
For Micronaut 2.0 and above you will need to add `@Executable` meta annotation to these annotations